Pcap++/CMakeLists.txt

```diff
@@ -24,6 +26,8 @@ add_library(
 
 set(public_headers
     header/Device.h
+    header/DeviceUtils.h
+    header/MemoryUtils.h
```

Comment:
These 2 new files contain internal utils, so they shouldn't be exposed as part of the API...

Reply:
Where do you want me to put them then?

Reply:
I'm not a CMake expert, but I think there is a way to decide which header files are "public" (meaning exposed as part of PcapPlusPlus API) and which are "private". This may also affect our doxygen API documentation: https://pcapplusplus.github.io/api-docs/next/ (which currently looks at all the header files in `Pcap++/header` 😕 )

Maybe @clementperon can help here?

Reply:
Well, removed the headers from the public header list. (fb685bf)

Things appear to be building fine, and the only difference is that the headers are missing from the include folder of the install command output.

Don't know if the docs generation has changed anything. The CI action still shows that the headers are being parsed.

Reply:
I think that's ok. For doxygen we probably need to exclude these files in [`Doxyfile`](https://github.com/PcapPlusPlus/pcapplusplus.github.io/blob/master/static/api-docs/next/doxygen/Doxyfile) and [`Doxyfile-ci`](https://github.com/PcapPlusPlus/pcapplusplus.github.io/blob/master/static/api-docs/next/doxygen/Doxyfile-ci) which are in the [web-site repo](https://github.com/PcapPlusPlus/pcapplusplus.github.io/). Here is how to exclude files:
https://stackoverflow.com/questions/34776315/doxygen-special-command-to-ignore-a-file

Do you want to open a PR for it or prefer I do it?

Reply:
Hmm, maybe we can mark the documentation as internal for doxygen in the code?

Something like this: https://stackoverflow.com/a/18443973/11922936
That way the docs won't be generated for the public documentation but it will retain the option for generating internal documentation if we want one in the future.

Reply:
Enclosed the internal docs in a conditional section (617273e). I think this might remove them from the public documentation.

Reply:
Nice!! You can test it locally or merge this PR and I'll test it later. It's not a blocker for merging because the documentation isn't final for a non-releases and we can fix it later if needed

Reply:
let's make it in another PR if necessary.